<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AudioInput</class>
 <widget class="QWidget" name="AudioInput">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1155</width>
    <height>713</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QPushButton" name="qpbAudioWizard">
     <property name="text">
      <string>Audio Wizard</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="qgbTransmission">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Transmission</string>
     </property>
     <layout class="QGridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="qliTransmit">
        <property name="text">
         <string>&amp;Transmit</string>
        </property>
        <property name="buddy">
         <cstring>qcbTransmit</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QComboBox" name="qcbTransmit">
        <property name="toolTip">
         <string>When to transmit your speech</string>
        </property>
        <property name="whatsThis">
         <string>&lt;b&gt;This sets when speech should be transmitted.&lt;/b&gt;&lt;br /&gt;&lt;i&gt;Continuous&lt;/i&gt; - All the time&lt;br /&gt;&lt;i&gt;Voice Activity&lt;/i&gt; - When you are speaking clearly.&lt;br /&gt;&lt;i&gt;Push To Talk&lt;/i&gt; - When you hold down the hotkey set under &lt;i&gt;Shortcuts&lt;/i&gt;.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="QStackedWidget" name="qswTransmit">
        <property name="currentIndex">
         <number>2</number>
        </property>
        <widget class="QWidget" name="qwPTT">
         <layout class="QGridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="qliDoublePush">
            <property name="text">
             <string>DoublePush Time</string>
            </property>
            <property name="buddy">
             <cstring>qsDoublePush</cstring>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QSlider" name="qsDoublePush">
            <property name="toolTip">
             <string>If you press the PTT key twice in this time it will get locked.</string>
            </property>
            <property name="whatsThis">
             <string>&lt;b&gt;DoublePush Time&lt;/b&gt;&lt;br /&gt;If you press the push-to-talk key twice during the configured interval of time it will be locked. RetroShare will keep transmitting until you hit the key once more to unlock PTT again.</string>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="singleStep">
             <number>10</number>
            </property>
            <property name="pageStep">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="qlDoublePush">
            <property name="text">
             <string notr="true">TextLabel</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="qwVAD">
         <layout class="QGridLayout" name="qwVadLayout_2">
          <item row="1" column="0">
           <widget class="QLabel" name="qliTransmitHold">
            <property name="text">
             <string>Voice &amp;Hold</string>
            </property>
            <property name="buddy">
             <cstring>qsTransmitHold</cstring>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QSlider" name="qsTransmitHold">
            <property name="toolTip">
             <string>How long to keep transmitting after silence</string>
            </property>
            <property name="whatsThis">
             <string>&lt;b&gt;This selects how long after a perceived stop in speech transmission should continue.&lt;/b&gt;&lt;br /&gt;Set this higher if your voice breaks up when you speak (seen by a rapidly blinking voice icon next to your name).</string>
            </property>
            <property name="minimum">
             <number>20</number>
            </property>
            <property name="maximum">
             <number>250</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="qliTransmitMin">
            <property name="text">
             <string>Silence Below</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QSlider" name="qsTransmitMin">
            <property name="toolTip">
             <string>Signal values below this count as silence</string>
            </property>
            <property name="whatsThis">
             <string>&lt;b&gt;This sets the trigger values for voice detection.&lt;/b&gt;&lt;br /&gt;Use this together with the Audio Statistics window to manually tune the trigger values for detecting speech. Input values below &quot;Silence Below&quot; always count as silence. Values above &quot;Speech Above&quot; always count as voice. Values in between will count as voice if you're already talking, but will not trigger a new detection.</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>32767</number>
            </property>
            <property name="singleStep">
             <number>100</number>
            </property>
            <property name="pageStep">
             <number>1000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="qliTransmitMax">
            <property name="text">
             <string>Speech Above</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QSlider" name="qsTransmitMax">
            <property name="toolTip">
             <string>Signal values above this count as voice</string>
            </property>
            <property name="whatsThis">
             <string>&lt;b&gt;This sets the trigger values for voice detection.&lt;/b&gt;&lt;br /&gt;Use this together with the Audio Statistics window to manually tune the trigger values for detecting speech. Input values below &quot;Silence Below&quot; always count as silence. Values above &quot;Speech Above&quot; always count as voice. Values in between will count as voice if you're already talking, but will not trigger a new detection.</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>32767</number>
            </property>
            <property name="singleStep">
             <number>100</number>
            </property>
            <property name="pageStep">
             <number>1000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="qlTransmitHold">
            <property name="minimumSize">
             <size>
              <width>40</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>empty</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="qwContinuous"/>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QGroupBox" name="qgbAudio">
       <property name="title">
        <string>Audio Processing</string>
       </property>
       <layout class="QGridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="qliNoise">
          <property name="text">
           <string>Noise Suppression</string>
          </property>
          <property name="buddy">
           <cstring>qsNoise</cstring>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSlider" name="qsNoise">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="toolTip">
           <string>Noise suppression</string>
          </property>
          <property name="whatsThis">
           <string>&lt;b&gt;This sets the amount of noise suppression to apply.&lt;/b&gt;&lt;br /&gt;The higher this value, the more aggressively stationary noise will be suppressed.</string>
          </property>
          <property name="minimum">
           <number>14</number>
          </property>
          <property name="maximum">
           <number>60</number>
          </property>
          <property name="pageStep">
           <number>5</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="qlNoise">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="qliAmp">
          <property name="text">
           <string>Amplification</string>
          </property>
          <property name="buddy">
           <cstring>qsAmp</cstring>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSlider" name="qsAmp">
          <property name="toolTip">
           <string>Maximum amplification of input sound</string>
          </property>
          <property name="whatsThis">
           <string>&lt;b&gt;Maximum amplification of input.&lt;/b&gt;&lt;br /&gt;RetroShare normalizes the input volume before compressing, and this sets how much it's allowed to amplify.&lt;br /&gt;The actual level is continually updated based on your current speech pattern, but it will never go above the level specified here.&lt;br /&gt;If the &lt;i&gt;Microphone loudness&lt;/i&gt; level of the audio statistics hover around 100%, you probably want to set this to 2.0 or so, but if, like most people, you are unable to reach 100%, set this to something much higher.&lt;br /&gt;Ideally, set it so &lt;i&gt;Microphone Loudness * Amplification Factor &gt;= 100&lt;/i&gt;, even when you're speaking really soft.&lt;br /&gt;&lt;br /&gt;Note that there is no harm in setting this to maximum, but RetroShare will start picking up other conversations if you leave it to auto-tune to that level.</string>
          </property>
          <property name="maximum">
           <number>19500</number>
          </property>
          <property name="singleStep">
           <number>500</number>
          </property>
          <property name="pageStep">
           <number>2000</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="qlAmp">
          <property name="minimumSize">
           <size>
            <width>30</width>
            <height>0</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QCheckBox" name="qcbEchoCancel">
          <property name="text">
           <string>Echo Cancellation Processing</string>
          </property>
          <property name="tristate">
           <bool>false</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Video Processing</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QVideoOutputDevice" name="videoDisplay">
        <property name="minimumSize">
         <size>
          <width>170</width>
          <height>128</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="voipGraph" name="voipBwGraph">
        <property name="sizePolicy">
         <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>1</width>
       <height>151</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QVideoOutputDevice</class>
   <extends>QFrame</extends>
   <header>gui/QVideoDevice.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>voipGraph</class>
   <extends>QFrame</extends>
   <header>gui/AudioInputConfig.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>qcbTransmit</tabstop>
  <tabstop>qsDoublePush</tabstop>
  <tabstop>qsTransmitHold</tabstop>
  <tabstop>qsTransmitMin</tabstop>
  <tabstop>qsTransmitMax</tabstop>
  <tabstop>qsNoise</tabstop>
  <tabstop>qsAmp</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
